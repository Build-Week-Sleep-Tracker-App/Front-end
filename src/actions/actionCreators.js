import types from './actionTypes';

export const login = () => {
	return {
		type: types.LOGIN
	}
}