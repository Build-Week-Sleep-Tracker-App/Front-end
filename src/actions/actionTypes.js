const LOGIN = 'LOGIN';
const SIGNUP = 'SIGNUP';

export default {
	LOGIN
}