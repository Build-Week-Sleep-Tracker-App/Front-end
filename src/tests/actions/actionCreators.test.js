import configureStore from 'redux-mock-store';

import { logout } from '../../actions/actionCreators';

const mockStore = configureStore();
const store = mockStore();

describe('test action creators', () => {
  beforeEach(() => {
    store.clearActions();
  });
  test('should dispatch correct logout action', () => {
    const expectedActions = [
      {
        type: 'LOGOUT'
      }
    ];

    store.dispatch(logout());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});
