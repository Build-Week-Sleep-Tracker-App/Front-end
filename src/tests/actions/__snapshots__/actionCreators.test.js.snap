// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test action creators should dispatch correct logout action 1`] = `
Array [
  Object {
    "type": "LOGOUT",
  },
]
`;
